### PR TITLE
Improve cheat execution parameters and handle CRLF properly

### PR DESCRIPTION
**Pull request by piepie62**

### CODE REVIEW

**Comment by piepie62** [comment]
This may fix #1212

File: sysmodules/rosalina/source/menus/cheats.c [hunk 1]
```diff
@@ -895,7 +893,13 @@ static Result Cheat_ReadLine(char* line)
         res = c ? 0 : -1;
         if (R_SUCCEEDED(res) && c != '\0')
         {
-            if (c == '\n' || c == '\r' || idx >= 1023)
+            if (c == '\r' && cheatFileBuffer[cheatFilePos] == '\n')
```
Comment (TuxSH) [inline sysmodules/rosalina/source/menus/cheats.c]: You could/should check if cheatFilePos doesn't overflow

**Review by TuxSH — APPROVED** [review]
Should be ok but someone that have tested the feature (ie. not me) should have a look at the key code changes.

File: sysmodules/rosalina/source/menus/cheats.c (hunk 1 above)
Comment (piepie62) [reply sysmodules/rosalina/source/menus/cheats.c]: Right, duh. Didn't think of that because it wasn't there in the first place. Fix incoming

**Comment by piepie62** [comment]
The key code changes are based on my understanding of the documentation; all of the other conditionals work on only the following block, but with the way it was implemented here, the conditional wouldn't even be properly checked, instead taking a shortcut that could be incorrect if there were multiple DD codes in the cheat.

**Comment by piepie62** [comment]
An example of where this could be a problem would be a code that has an enable and a disable keycode

**Pull request merged**